plugins/woocommerce/src/Blocks/BlockTypes/AddToCartWithOptions.php

```diff
@@ -68,12 +68,18 @@ protected function render( $attributes, $content, $block ) {
 
 		$previous_product = $product;
 		$product          = wc_get_product( $post_id );
-		if ( ! $product instanceof \WC_Product ) {
+		if ( ! $product instanceof \WC_Product || ( ProductType::SIMPLE === $product->get_type() && ! $product->is_purchasable() ) ) {
 			$product = $previous_product;
 
 			return '';
 		}
 
+		if ( ! $product->is_in_stock() ) {
+			$product = $previous_product;
+
+			return wc_get_stock_html( $product );
+		}
```

Comment:
I'm sorry for the incremental review, but after comparing the behavior introduced in this PR and the classic add-to-cart template, I think we should cover not only out-of-stock status but also backorder an other statuses.

The way it works for the classic template is:
- always displaying the availability text.
- displaying the add-to-cart form conditionally based on stock status.

Maybe we should create a block for availability text?

Reply:
> I'm sorry for the incremental review

Not at all, thanks for noticing it! :pray: 

Instead of always displaying the Stock text, I added the _Product Stock Indicator_ block to the simple product template part (2da6ff7e383840ae8534a44199022c3c73b2da4e).

I also took the chance to update the Grouped Product template part to display when child products are out of stock (d0d1c2a625c644f3c2e99b0943dd235d028fa4e6). I updated the PR testing steps accordingly.